app/controllers/api/v2/config_reports_controller.rb

```diff
@@ -40,7 +40,10 @@ def show
 
       def create
         @config_report = ConfigReport.import(params.to_unsafe_h[:config_report], detected_proxy.try(:id))
-        process_response @config_report.errors.empty?
+        errors = @config_report.errors.empty?
+        # RABL is extremely slow for huge reports and ENC script does not care about the output
+        @config_report = {}
```

Comment:
could we at least return report id? that way API consumer knows, how to load additional data

Reply:
What if instead of having [`create.json.rabl`](https://github.com/theforeman/foreman/blob/develop/app/views/api/v2/config_reports/create.json.rabl) do `extends "api/v2/config_reports/show"` (which contains the full report info) we change it to `extends "api/v2/config_reports/main"` or `extends "api/v2/config_reports/base"` which are much more minimal?

Reply:
Absolutely, I will do that. Also I will look into optimizing show.rabl.